### PR TITLE
Update cuco git tag to fetch new multiset data structure

### PR DESCRIPTION
**Pull request by PointKernel**
## Description
This PR bumps the cuco version to fetch the new multiset data structure.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/rapids-cmake/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The `cmake-format.json` is up to date with these changes.
- [ ] I have added new files under rapids-cmake/
   - [ ] I have added include guards (`include_guard(GLOBAL)`)
   - [ ] I have added the associated docs/ rst file and update the api.rst

### CODE REVIEW

File: rapids-cmake/cpm/versions.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
       "version": "0.0.1",
       "git_shallow": false,
       "git_url": "https://github.com/NVIDIA/cuCollections.git",
-      "git_tag": "2101cb31d0210b609cd02c88f9b538e10881d91d"
+      "git_tag": "03304a3e204d65783c90c8dd212f8aaf20df0efb"
```
Comment (PointKernel) [inline rapids-cmake/cpm/versions.json:39]:
```suggestion
      "git_tag": "47766b33283e7646df14286481780abbe112fc11"
```

File: rapids-cmake/cpm/versions.json [hunk 2]
```diff
@@ -23,7 +23,7 @@
       "git_shallow": false,
       "always_download": true,
       "git_url": "https://github.com/NVIDIA/cuCollections.git",
-      "git_tag": "6923b3b1e526b3305de81f784fdca26736caf5ba"
+      "git_tag": "47766b33283e7646df14286481780abbe112fc11"
```
Comment (PointKernel) [inline rapids-cmake/cpm/versions.json:26]:
```suggestion
      "git_tag": "369e7dacc15d7641c671af4d17455bd52eff0c98"
```

**Comment by PointKernel** [comment]
Corresponding PRs in each project for build verification:

- https://github.com/rapidsai/cudf/pull/15938
- https://github.com/rapidsai/raft/pull/2372
- https://github.com/rapidsai/cugraph/pull/4512

**Comment by PointKernel** [comment]
All repos passed CI checks without failures.

I'm not sure if the conda cpp test failure is related or not.

**Comment by PointKernel** [comment]
Using a different cuco git tag seems to solve the CI failures.

Let me put this as a draft PR to test the new cuco git tag against relevant RAPIDS projects.

DO NOT MERGE

**Comment by PointKernel** [comment]
I wanted to fetch the latest SHA of cuco since it includes a bug fix but that git tag failed the CI again. @robertmaynard can you please take a look?

**Comment by robertmaynard** [comment]
Ci is all passing now.

**Review by robertmaynard — APPROVED** [review]

**Comment by PointKernel** [comment]
cudf CI passed but some of the raft and cugraph CI tasks are still failing. I will ping here once all CI passes.

**Comment by PointKernel** [comment]
All repos passed. This is ready to merge.

**Review by bdice — APPROVED** [review]

**Pull request merged**